dashboard/test/ui/features/star_labs/aichat/chat.feature

```diff
@@ -47,6 +47,7 @@ Feature: AI Chat
     And I wait until element "#uitest-publish-notes-save" is enabled
     And I click selector "#uitest-publish-notes-save"
     Then I wait until element ".uitest-aichat-chat-alert" contains text "Model card information has been updated"
+    And I wait for 5 seconds
```

Comment:
Updated to 5 seconds. Was successful at [2 secs](https://app.saucelabs.com/tests/99f94b5266c34fffae096ea9c9770dd4#111) when I ran locally so a bit befuddled on why failing on Drone.

Reply:
@bencodeorg shared [this Slack thread](https://codedotorg.slack.com/archives/C03CK49G9/p1725476577054369?thread_ts=1725475362.107969&cid=C03CK49G9
) - failure is most likely due to Drone not set up to have access to AWS Comprehend.
Investigating now why previous scenario is passing (saving system prompt) while this one is failing. We've seen different behavior between the first and subsequent saving of sources.

Reply:
Woo hoo! Figured out why the prior scenario when the user edits then saves the system prompt does not fail on Drone.
In `SourcesStore.ts` in `save`, [we check for `this.currentVersionId`](https://github.com/code-dot-org/code-dot-org/blob/4dbbb2ab334ebc37d68a852b7083b3f2232937dc/apps/src/lab2/projects/SourcesStore.ts#L65-L90), if not defined, then `options` remains `undefined` . Thus, `projectType` is `undefined` so that in `files_api` in `put_file`, the call to `AichatComprehendHelper` is never made. 
https://github.com/code-dot-org/code-dot-org/blob/fd596c0ba6bcd071b09b4715ca9d661e5f9e3a92/dashboard/legacy/middleware/files_api.rb#L426


Thus, the system prompt and retrievals are not checked for toxicity the first time sources are saved, but are checked on subsequent saves which is why the scenario for publishing model card info was failing. The user saved, then published the model card and publishing includes saving sources.

Reply:
I pushed a commit to skip the publish model card scenario.
I will file a ticket to fix this saving bug (we should be checking for toxicity on the first time!) and in that follow-up, we'll need to either skip the other scenario (editing/saving system prompt) or perhaps stub calls to AWS Comprehend on Drone as @bencodeorg has been looking into.

Reply:
Great find!